eclair-core/src/main/scala/fr/acinq/eclair/api/EclairApi.scala

```diff
@@ -0,0 +1,224 @@
+package fr.acinq.eclair.api
+
+import akka.util.Timeout
+import akka.pattern._
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.bitcoin.{ByteVector32, MilliSatoshi, Satoshi}
+import fr.acinq.eclair.{Kit, ShortChannelId}
+import fr.acinq.eclair.io.{NodeURI, Peer}
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.db.{NetworkFee, Stats}
+import fr.acinq.eclair.io.Peer.{GetPeerInfo, PeerInfo}
+import fr.acinq.eclair.payment.PaymentLifecycle._
+import fr.acinq.eclair.payment.{PaymentLifecycle, PaymentReceived, PaymentRequest}
+import fr.acinq.eclair.router.{ChannelDesc, RouteNotFound, RouteRequest, RouteResponse}
+import fr.acinq.eclair.wire.{ChannelAnnouncement, ChannelUpdate, NodeAddress, NodeAnnouncement}
+import scodec.bits.ByteVector
+import scala.concurrent.duration._
+import scala.concurrent.Future
+
+trait EclairApi {
+
+  def connect(uri: String): Future[String]
+
+  def open(nodeId: PublicKey, fundingSatoshis: Long, pushMsat: Option[Long], fundingFeerateSatByte: Option[Long], flags: Option[Int]): Future[String]
+
+  def close(channelIdentifier: Either[ByteVector32, ShortChannelId], scriptPubKey: Option[ByteVector]): Future[String]
+
+  def forceClose(channelIdentifier: Either[ByteVector32, ShortChannelId]): Future[String]
+
+  def updateRelayFee(channelId: String, feeBaseMsat: Long, feeProportionalMillionths: Long): Future[String]
+
+  def peersInfo(): Future[Iterable[PeerInfo]]
+
+  def channelsInfo(toRemoteNode: Option[PublicKey]): Future[Iterable[RES_GETINFO]]
+
+  def channelInfo(channelId: ByteVector32): Future[RES_GETINFO]
+
+  def allnodes(): Future[Iterable[NodeAnnouncement]]
+
+  def allchannels(): Future[Iterable[ChannelDesc]]
+
+  def allupdates(nodeId: Option[PublicKey]): Future[Iterable[ChannelUpdate]]
+
+  def receive(description: String, amountMsat: Option[Long], expire: Option[Long]): Future[String]
+
+  def findRoute(targetNodeId: PublicKey, amountMsat: Long, assistedRoutes: Seq[Seq[PaymentRequest.ExtraHop]] = Seq.empty): Future[RouteResponse]
+
+  def send(recipientNodeId: PublicKey, amountMsat: Long, paymentHash: ByteVector32, assistedRoutes: Seq[Seq[PaymentRequest.ExtraHop]] = Seq.empty): Future[PaymentResult]
+
+  def checkpayment(paymentHash: ByteVector32): Future[Boolean]
+
+  def audit(from_opt: Option[Long], to_opt: Option[Long]): Future[AuditResponse]
+
+  def networkFees(from_opt: Option[Long], to_opt: Option[Long]): Future[Seq[NetworkFee]]
+
+  def channelStats(): Future[Seq[Stats]]
+
+  def help(): List[String]
```

Comment:
`help` should'n be part of the programmatic API

Reply:
Done via 1fe65f4